### PR TITLE
MRG, ENH: Make _get_hpi_info public

### PR DESCRIPTION
**Pull request by hoechenberger**
I need this functionality in MNE-BIDS to enrich our MEG sidecar files with information about cHPI coil frequencies. Therefore I decided to make this function public.

WIP: Need to add tests for the newly supported data types

cc @agramfort

### CODE REVIEW

File: doc/preprocessing.rst [hunk 1]
```diff
@@ -156,6 +156,7 @@ EEG referencing:
 .. autosummary::
    :toctree: generated/
 
+   get_hpi_info
```
Comment (hoechenberger) [inline doc/preprocessing.rst:159]: Is this supposed to be in alphabetic order here?

File: tutorials/preprocessing/59_head_positions.py [hunk 2]
```diff
@@ -11,6 +11,11 @@
 of time. Here we show an example of how to do this, and how to visualize
 the result.
 
+Authors: Eric Larson <larson.eric.d@gmail.com>
+         Richard Höchenberger <richard.hoechenberger@gmail.com>
+
+License: BSD (3-clause)
+
```
Comment (agramfort) [inline tutorials/preprocessing/59_head_positions.py:18]: we add this after the __doc__ header usually.

File: tutorials/preprocessing/59_head_positions.py [hunk 3]
```diff
@@ -20,9 +25,6 @@
 frequencies, plus other peaks related to power line interference (60 Hz and
 harmonics).
 """
-# Authors: Eric Larson <larson.eric.d@gmail.com>
-#
-# License: BSD (3-clause)
```
Comment (agramfort) [inline tutorials/preprocessing/59_head_positions.py:25]: it was at the right place

File: doc/changes/latest.inc [hunk 4]
```diff
@@ -23,7 +23,7 @@ Enhancements
 ~~~~~~~~~~~~
 .. - Add something cool (:gh:`9192` **by new contributor** |New Contributor|_)
 
-- Nothing yet
+- New function :func:`mne.chpi.get_chpi_info` to retrieve basic information about the cHPI system used when recording MEG data (this functionality was hidden in a private function before) (:gh:`9369` by `Richard Höchenberger`_)
```
Comment (larsoner) [inline doc/changes/latest.inc:26]: Users shouldn't ever need to care or think about or private functions (I would say it's better if they don't know of their existence, lest they get curious about what other functionality we're hiding :) )
```suggestion
- New function :func:`mne.chpi.get_chpi_info` to retrieve basic information about the cHPI system used when recording MEG data (:gh:`9369` by `Richard Höchenberger`_)
```

File: mne/chpi.py [hunk 5]
```diff
@@ -295,14 +295,51 @@ def extract_chpi_locs_kit(raw, stim_channel='MISC 064', *, verbose=None):
...
+    from .io import BaseRaw
+    from . import BaseEpochs
+    from .evoked import Evoked
+
+    _validate_type(item=inst, item_name='inst',
+                   types=(Info, BaseRaw, BaseEpochs, Evoked))
+    _check_option(parameter='on_missing', value=on_missing,
+                  allowed_values=['ignore', 'raise'])
+
+    info = inst if isinstance(inst, Info) else inst.info
```
Comment (larsoner) [inline mne/chpi.py:336]: I think I would rather just have people pass `inst.info` here

File: mne/chpi.py [hunk 6]
```diff
@@ -295,14 +295,51 @@ def extract_chpi_locs_kit(raw, stim_channel='MISC 064', *, verbose=None):
...
+    """Retrieve cHPI information from the data.
+
+    Parameters
+    ----------
+    inst : instance of Info | Raw | Epochs | Evoked
+        The object from which to extract the HPI information.
+    on_missing : 'ignore' | 'raise'
+        What to do if no HPI information can be found. If ``'ignore'``
+        (default), all return values will be empty arrays or ``None``. If
+        ``'raise'``, an exception will be raised.
```
Comment (larsoner) [inline mne/chpi.py:310]: Better / more DRY to use `on_missing_base` in `docs.py` (see how we do it for other `on_*` in other functions) and the helper `mne.utils. _on_missing`. I don't see a reason why `'warn'` shouldn't be supported here.

File: mne/chpi.py (hunk 5 above)
Comment (hoechenberger) [reply mne/chpi.py:336]: I thought supporting more object types could provide slightly more convenience for users … no? :)

Comment (agramfort) [reply mne/chpi.py:336]: I would stick to passing info.

opening more options means more tests, more documentation etc.

**Comment by hoechenberger** [comment]
I believe I have addressed all of the review comments

**Comment by larsoner** [comment]
Failures look legitimate

**Comment by larsoner** [comment]
Thanks @hoechenberger !

**Pull request merged**